From: Modestas Vainius <modax@debian.org>
SUbject: Make hide_symbols to cover all QtWebKit portions
Origin: vendor
Last-Update: 2012-05-12
Forwarded: no

Almost everything in Source/WebCore/generated was not build with
-fvisibility=hidden resulting into redundantly polluted and huge dynamic symbol
list in libQtWebKit.so.

Therefore, move +hide_symbols configuration parameter upwards in order for it
to cover everything including the offending Source/WebCore/CodeGenerators.pri

--- a/Source/WebKit/qt/QtWebKit.pro
+++ b/Source/WebKit/qt/QtWebKit.pro
@@ -15,6 +15,11 @@ else: CONFIG_DIR = release
 
 SOURCE_DIR = $$replace(PWD, /WebKit/qt, "")
 
+CONFIG(release):!CONFIG(standalone_package) {
+    contains(QT_CONFIG, reduce_exports):CONFIG += hide_symbols
+    unix:contains(QT_CONFIG, reduce_relocations):CONFIG += bsymbolic_functions
+}
+
 include($$PWD/Api/headers.pri)
 include($$SOURCE_DIR/WebKit.pri)
 include($$SOURCE_DIR/JavaScriptCore/JavaScriptCore.pri)
@@ -66,11 +71,6 @@ defineTest(addExtraCompiler) {
 
 include($$SOURCE_DIR/WebCore/CodeGenerators.pri)
 
-CONFIG(release):!CONFIG(standalone_package) {
-    contains(QT_CONFIG, reduce_exports):CONFIG += hide_symbols
-    unix:contains(QT_CONFIG, reduce_relocations):CONFIG += bsymbolic_functions
-}
-
 CONFIG(QTDIR_build) {
     include($$QT_SOURCE_TREE/src/qbase.pri)
     # The following lines are to prevent qmake from adding the jscore, webcore and webkit2 libs to libQtWebKit's prl dependencies.
